txp/Pos/StateLock/PrioLock.hs

```diff
@@ -0,0 +1,77 @@
+{-|
+Module:      Pos.StateLock.PrioLock
+Description: Provides a prioritised lock
+
+Provides a lock that can be taken with either high or low precedence.
+Within each precedence, the lock is taken in FIFO order.
+-}
+
+{-# LANGUAGE LambdaCase #-}
+
+module Pos.StateLock.PrioLock where
```

Comment:
I think this module can be even in `cardano-sl-core` somewhere in `Pos.Util` (maybe `Pos.Util.Concurrent.PrioLock`).
It doesn't depend on anything from CSL and it's quite general, not related to transaction processing at all.

Reply:
Yes, that's a good place to put this.